src/hoc/ClientSideStateUpdate.tsx

```diff
@@ -81,7 +83,11 @@ export const ClientSideStateUpdate = ({
     if (assigneeSuggestions) {
       store.dispatch(setAssigneeSuggestion(assigneeSuggestions))
     }
-  }, [workflowStates, tasks, token, assignee, viewSettings, tokenPayload, templates, assigneeSuggestions])
+    if (task) {
+      const updatedTasks = tasksInStore.map((t) => (t.id === task.id ? task : t))
```

Comment:
Is there a reason for doing this? Why not just use tasksInStore directly instead?

Is something other than `id` changed between `task` and `t`?